spec/app/tournament/tournament.spec.js

```diff
@@ -0,0 +1,145 @@
+import * as player from "../../../app/tournament/player";
+import * as tournament from "../../../app/tournament/tournament";
+
+describe('class Tournament', function() {
+    it('adding a second players with same name should throw an error', function() {
+        var mPlayer = new player.Player("pepe", 2);
+        var mTournament = new tournament.Tournament();
+        mTournament.addPlayer(mPlayer);
+        expect( function() {mTournament.addPlayer(mPlayer)} ).toThrow(new tournament.TournamentError(`Duplicate player: "${player.name}".`));
+    });
+
+    it('generating a tournament with less than 2 players throw an error', function() {
+        var mTournament = new tournament.Tournament();
+        expect( function() {mTournament.generate()} ).toThrow(new tournament.TournamentError(`Too few players: ${mTournament.players.length}`));
+
+        mTournament.addPlayer(new player.Player("Spiderman", 1));
+        expect( function() {mTournament.generate()} ).toThrow(new tournament.TournamentError(`Too few players: ${mTournament.players.length}`));
+    });
+
+    it('a total of 3 players should not be a proper number', function() {
+        var mTournament = new tournament.Tournament();
+        mTournament.addPlayer(new player.Player("Spiderman", 1));
+        mTournament.addPlayer(new player.Player("Batman", 1));
+        mTournament.addPlayer(new player.Player("Ironman", 1));
```

Comment:
Is it valid to have 3 players with the same rank? It´s not the best way to specify your algorithm.